descriptors/ModuleDescriptor-template.json

```diff
@@ -1,6 +1,28 @@
 {
   "id": "${artifactId}-${version}",
   "name": "Circulation Storage Module",
+  "requires": [
+    {
+      "id": "login",
+      "version": "7.0"
+    },
```

Comment:
Is this needed because the request expiration process is not an OKAPI timer event but Vert.x periodic task, so we don't have neither okapi-url nor user-token?

If yes, then we should just convert the expiration process to an OKAPI scheduled job. What do you think?

Reply:
Yes, you are correct. Could you please provide more details on OKAPI scheduled job?

Reply:
Do you mean timer interface https://github.com/folio-org/okapi/blob/master/doc/guide.md#timer-interface?

Reply:
@bohdan-suprun @marcjohnson-kint  I tried to minimize interference with the existing logic. If you guys think that timer interface will be more suitable then I will redo this PR. Let me please know your opinion.

Reply:
Yes, I've pushed some draft changes for this, see [CIRCSTORE-247-use-okapi-scheduled-jobs-for-rq-expiration branch](https://github.com/folio-org/mod-circulation-storage/compare/CIRCSTORE-247-use-okapi-scheduled-jobs-for-rq-expiration?expand=1)

Reply:
I think it would be preferable to use the okapi timer interface. The built in RAML Module Builder timed execution is pretty much deprecated and contributes to some operational challenges.

I don't think it is appropriate to impersonate the `pub-sub` user during these processes (it has sometimes been an unfortunate side effect of using pub-sub in other processes due the how message delivery works). I understand it is convenient because that user already exists. It creates a subtle coupling between this module and mod-pubsub.

If this code is switched to use the okapi timer interface, this process will effectively be anonymous, meaning there will not be a user associated with any of these log entries.

In general, I think that storage modules should try to avoid dependencies wherever possible. (this is based upon my understanding of the original intent of storage modules, that I'm trying to verify)

My personal preference would've been for these activities to be moved to mod-circulation and then extends to publish messages for the circulation log. I imagine we aren't in a position to do that at this point.

Mikhail sought a special case for a dependency on `Kafka` from `mod-inventory-storage` yesterday and `mod-source-record-storage` already has dependencies on `pub-sub` so the dependencies are creeping in anyway.

Reply:
OK, guys. I start on refactoring. Thank you!